### PR TITLE
OSSM-4390: sync Route informers before starting IOR controller

### PR DESCRIPTION
**Pull request by yannuil**
**Please provide a description of this PR:**

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Ambient
- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [ ] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

**Comment by jewertow** [comment]
Could you explain this change? I don't understand what actually changes.

**Comment by yannuil** [comment]
IOR uses `pilot.pkg.config.kube.crdclient` to manage the route informer's events. According to [this comment](https://github.com/maistra/istio/blob/maistra-2.4/pkg/kube/client.go#L606), after [getting the `Route` informer](https://github.com/maistra/istio/blob/maistra-2.4/pilot/pkg/config/kube/ior/controller.go#L79) in the IOR we need to start and wait for the informer cache sync. This changes make sure that the start and wait run after the calling the `.RouteInformer()` method.

File: pilot/pkg/bootstrap/configcontroller.go [hunk 1]
```diff
@@ -225,7 +225,11 @@ func (s *Server) initK8SConfigStore(args *PilotArgs) error {
 			leaderelection.
 				NewLeaderElection(args.Namespace, args.PodName, leaderelection.IORController, args.Revision, s.kubeClient).
 				AddRunFunction(func(leaderStop <-chan struct{}) {
-					ior.Run(s.kubeClient, configController, leaderStop)
+					controller := ior.New(s.kubeClient, configController)
+
+					s.kubeClient.RunAndWaitRouteInformer(leaderStop)
+
+					controller.Run(leaderStop)
```
Comment (jewertow) [inline pilot/pkg/bootstrap/configcontroller.go:232]: Could not it be simple like this?
```suggestion
                                        s.kubeClient.RunAndWait(stop)
					ior.Run(s.kubeClient, configController, leaderStop)
```
Then you could revert all other changes from this PR.

**Comment by FilipB** [comment]
See attached reproducer in https://issues.redhat.com/browse/OSSM-5720 which will simplify the testing

**Comment by FilipB** [comment]
What's the status? Do we want to test this before merge?

File: pilot/pkg/bootstrap/configcontroller.go (hunk 1 above)
Comment (jewertow) [reply pilot/pkg/bootstrap/configcontroller.go:232]: Ok, now I see that it couldn't work, because `RunAndWait()` must be called after `.Informer()`.

**Review by jewertow — APPROVED** [review]

**Pull request merged**